dimod/bqm/src/shapeable.cc

```diff
@@ -0,0 +1,315 @@
+// Copyright 2019 D-Wave Systems Inc.
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+#include "src/adjarray.h"
+#include "src/shapeable.h"
```

Comment:
Again, function templates should be defined in the header where they are declared. I'm not sure what cython is doing, but I think this would be needed to use this as a component of a C++ project.
See: https://google.github.io/styleguide/cppguide.html#Self_contained_Headers